hash_practice/exercises.py

```diff
@@ -1,20 +1,53 @@
 
+import numbers
+
+
 def grouped_anagrams(strings):
-    """ This method will return an array of arrays.
-        Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+    """ 
+        Time Complexity: O(n log(n))
+        Space Complexity: O(n)
     """
-    pass
+    anagram_dict = {}
+
+    for word in strings: 
+        list_of_letters = list(word)
+        list_of_letters.sort()
+        key = "".join(list_of_letters) 
+
+        if word not in anagram_dict:
+            anagram_dict[key] = anagram_dict.get(key,[])

```

Comment:
Ah, I think I notice a small "bug", but not one big enough to mark off points.

At first glance, you shouldn't need to use `get` here because you checked the dictionary already, right? But since you're checking if `word` is in the dictionary rather than `key` in line 17, this probably had to be added since `word` will only be in the dictionary if all of its letters are sorted (because in that case it'll be equal to `key`).

Not a big deal, I only point it out in case you were wondering what happened here.